File: 3-swagger-api-documentation/0-typescript/src/modules/cat/dto/create-cat.dto.ts
/**
 * DTO validates creation payload.
 */
import {
    ApiProperty 
} from "@nestjs/swagger"
import {
    Type 
} from "class-transformer"
import {
    IsNumber, IsString 
} from "class-validator"

/** create-cat DTO; Swagger needs runtime metadata from `@ApiProperty`.
 */
export class CreateCatDto {
    @ApiProperty({
        example: "Persian",
        description: "Breed of the cat",
    })
    @IsString()
        breed!: string

    @ApiProperty({
        example: 2,
        description: "Age in years",
    })
    @Type(() => Number)
    @IsNumber()
        age!: number
}

File: 3-swagger-api-documentation/1-java/src/main/java/academy/starci/swagger/CreateCatDto.java
package academy.starci.swagger;

import io.swagger.v3.oas.annotations.media.Schema;

public class CreateCatDto {

    @Schema(example = "Persian", description = "Breed of the cat")
    private String breed;

    @Schema(example = "2", description = "Age in years")
    private Integer age;

    public String getBreed() {
        return breed;
    }

    public void setBreed(String breed) {
        this.breed = breed;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }
}
